ddcommon-ffi/src/array_queue.rs

```diff
@@ -0,0 +1,294 @@
+// Copyright 2021-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::Error;
+use anyhow::Context;
+use std::ffi::c_void;
+
+#[repr(C)]
+// A simple wrapper around crossbeam_queue::ArrayQueue<*mut c_void>, which is a lock free
+// bounded multi-producer and multi-consumer (MPMC) queue.
+pub struct ArrayQueue {
```

Comment:
Since you implement drop, you might need a phantom data

Reply:
Would miri complain about this?

Reply:
Here the ArrayQueue only performs push and pop, don't have peek() which could have required me to have a phantom data.